File: bot-frontend/src/features/rpc-groups/types/rpcGroup.ts
export interface RPCGroup {
  id: number;
  name: string;
  group: string;
  creation_time: string;
}

export interface RPC {
  http: string;
  ws: string;
}

export interface RPCGroupDashboardRow {
  id: number;
  name: string;
  number: number;
  creation_time: string;
}

export interface RPCGroupPost {
  name: string;
  group: string;
}

export interface RPCGroupPut {
  id: number;
  name: string;
  group: string;
}

File: backend/api/dto/rpc_group.go
package dto

type RPCGroupDashboard []RPCGroupDashboardRow

type RPCGroupDashboardRow struct {
	Id           int64  `json:"id"`
	Name         string `json:"name"`
	GroupCount   int    `json:"count"`
	CreationTime string `json:"creation_time"`
}

type RPCGroup struct {
	Id           int64   `json:"id"`
	Name         string  `json:"name"`
	Group        []Group `json:"group"`
	CreationTime string  `json:"creation_time"`
}

type RPCGroupResponse struct {
	Id           int64  `json:"id"`
	Name         string `json:"name"`
	Group        string `json:"group"`
	CreationTime string `json:"creation_time"`
}

type Group struct {
	Http string `json:"http"`
	Ws   string `json:"ws"`
}

// we will process the string internally in b.e.
type RPCGroupPush struct {
	Name  string `json:"name"`
	Group string `json:"group"`
}
